x-pack/plugins/security_solution/public/common/components/event_details/event_fields_browser.tsx

```diff
@@ -152,14 +172,54 @@ export const EventFieldsBrowser = React.memo<Props>(
       ]
     );
 
+    const focusSearchInput = useCallback(() => {
+      // the selector below is used to focus the input because EuiInMemoryTable does not expose a ref to its built-in search input
+      containerElement.current?.querySelector<HTMLInputElement>('input[type="search"]')?.focus();
+    }, [containerElement]);
+
+    const focusAddTimelineButton = useCallback(() => {
+      // the document selector below is required because we may be in a flyout or full screen timeline context
+      document.querySelector<HTMLButtonElement>(`.${ADD_TIMELINE_BUTTON_CLASS_NAME}`)?.focus();
+    }, []);
+
+    const onKeyDown = useCallback(
+      (keyboardEvent: React.KeyboardEvent) => {
+        if (isTab(keyboardEvent)) {
+          onEventDetailsTabKeyPressed({
+            containerElement: containerElement.current,
+            keyboardEvent,
+            onSkipFocusBeforeEventsTable: focusSearchInput,
+            onSkipFocusAfterEventsTable: focusAddTimelineButton,
+          });
+        } else {
+          onKeyDownFocusHandler({
+            colindexAttribute: DATA_COLINDEX_ATTRIBUTE,
+            containerElement: containerElement?.current,
+            event: keyboardEvent,
+            maxAriaColindex: 3,
+            maxAriaRowindex: data.length,
+            onColumnFocused: noop,
+            rowindexAttribute: DATA_ROWINDEX_ATTRIBUTE,
+          });
+        }
+      },
+      [containerElement, data, focusAddTimelineButton, focusSearchInput]
```

Comment:
containerElement is mutable values like 'ref.current',they aren't valid dependencies because mutating them doesn't re-render the component.